src/main/java/net/pms/util/ImagesUtil.java

```diff
@@ -145,4 +152,22 @@ public static void parseImageByImaging(File file, DLNAMediaInfo media) throws Im
 			media.setCodecV(FormatConfiguration.TIFF);
 		}
 	}
+
+	public static InputStream addFormatLabelToImage(BufferedImage image, String label, String outputImageFormat) throws IOException {
+		// copy image to not affect the original one
+		ColorModel cm = image.getColorModel();
+		boolean isAlphaPremultiplied = cm.isAlphaPremultiplied();
+		WritableRaster raster = image.copyData(null);
+		BufferedImage img = new BufferedImage(cm, raster, isAlphaPremultiplied, null);
+
+		Graphics2D g = img.createGraphics();
+		g.drawImage(img, 0, 0, null);
+		g.setColor(Color.WHITE);
+		g.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 15));
+        g.drawString(label.toUpperCase(), 30, 30);
+		ByteArrayOutputStream out = new ByteArrayOutputStream();
+		ImageIO.write(img, outputImageFormat, out);
+		g.dispose();
+		return new ByteArrayInputStream(out.toByteArray());
+	}
```

Comment:
I can't leave a comment directly on an image so I'm leaving it here - they look good but I think we should use images with transparent backgrounds only. Images with white backgrounds will look wrong on a renderer with a dark theme.